src/Psl/Iter/reverse.php

```diff
@@ -19,12 +19,17 @@
  * @psalm-param    iterable<T> $iterable The iterable to reverse.
  *
  * @psalm-return   Iterator<int, T>
+ *
+ * @deprecated since 1.2, use Vec\reverse instead.
+ *
+ * @see Vec\reverse()
  */
 function reverse(iterable $iterable): Iterator
 {
     return Iterator::from(static function () use ($iterable): Generator {
-        $values = to_array($iterable);
-        $size   = Arr\count($values);
+        $values = Vec\values($iterable);
+        $size   = namespace\count($values);
```

Comment:
Any specific reason to add the explicit namespace keyword here? Doesn't it fall back to the count in namespace anyways?

Reply:
so that if Iter\count wasn't loaded ( assuming someone is trying to require_once the files themselves instead of using the loader ), it will fail rather than falling back to the global count.

generally, we want to avoid this weird behavior in PHP were it falls back to using another function declaration when one is not found, so we should use either `Component\func`, `namespace\func`, or `\func` for global functions ( we don't enforce this currently, but we probably should ).